src/sentry/static/sentry/app/views/settings/components/forms/projectMapperField.tsx

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+import styled from '@emotion/styled';
+import {components} from 'react-select';
+
+import InputField from 'app/views/settings/components/forms/inputField';
+import {ProjectMapperType} from 'app/views/settings/components/forms/type';
+import SelectControl from 'app/components/forms/selectControl';
+import IdBadge from 'app/components/idBadge';
+import Button from 'app/components/button';
+import {IconAdd} from 'app/icons';
+import {t} from 'app/locale';
+
+type Props = InputField['props'];
+type RenderProps = Props & ProjectMapperType;
+
+export default class ProjectMapperField extends React.Component<Props> {
```

Comment:
Should this be a class if we're not using state?

Reply:
@EvanPurkhiser Yea, for the refs